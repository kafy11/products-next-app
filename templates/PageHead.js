import Head from 'next/head';

//Template for the head of the document adding title and favicon
const PageHead = ({ title, favicon = "/favicon.ico" }) => (
    <Head>
        <title>{title}</title>
        <link rel="icon" href={favicon} />
    </Head>
);

export default PageHead;