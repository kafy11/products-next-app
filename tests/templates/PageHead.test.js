import { shallow } from 'enzyme';
import PageHead, { defaultFavicon } from '../../templates/PageHead';

const title = "Test";

//function to help test favicon href
const expectFaviconHref = (wrapper, href) => {
    expect(wrapper.find('.page-head__favicon').prop('href')).toBe(href);
};

describe('PageHead', () => {
    it('should render correctly', () => {
        const wrapper = shallow(<PageHead title={title} />);
        expect(wrapper).toMatchSnapshot();
    });
    
    it('should render the title correctly', () => {
        const wrapper = shallow(<PageHead title={title} />);
        expect(wrapper.find('title').text()).toBe(title);
    });
    
    it('should render the favicon correctly', () => {
        const favicon = "Icone teste";
        const wrapper = shallow(<PageHead title={title} favicon={favicon} />);
        expectFaviconHref(wrapper, favicon);
    });
    
    it('should render the default favicon correctly', () => {
        const wrapper = shallow(<PageHead title={title} />);
        expectFaviconHref(wrapper, defaultFavicon);
    });
});