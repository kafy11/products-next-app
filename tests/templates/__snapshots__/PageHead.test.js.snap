// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHead should render correctly 1`] = `
<Head>
  <title>
    Test
  </title>
  <link
    className="page-head__favicon"
    href="/favicon.ico"
    rel="icon"
  />
</Head>
`;
